src/lib.rs

```diff
@@ -0,0 +1,54 @@
+use serde_json_core;
+
+pub trait StringSet {
+    fn string_set(&mut self, topic_parts:
+            core::iter::Peekable<core::str::Split<char>>, value: &str) ->
+            Result<(),()>;
+
+}
+
+macro_rules! derive_primitive {
```

Comment:
Why do we need to derive this for primitives? Aren't primitives always going to be the base-case? Is this because of the condition where we request e.g. "a/b", where "a" is already a primitive? If so, should outline that.